src/ResourceManager/RecoveryServices.SiteRecovery/Commands.RecoveryServices.SiteRecovery/Vault/SetAzureRmRecoveryServicesAsrVaultContext.cs

```diff
@@ -0,0 +1,126 @@
+
+using System;
+using System.Net;
+using System.Management.Automation;
+using Microsoft.Rest.Azure;
+using Microsoft.Azure.Portal.RecoveryServices.Models.Common;
+using Microsoft.Azure.Commands.RecoveryServices.SiteRecovery.Properties;
+using Microsoft.Azure.Management.RecoveryServices.Models;
+
+namespace Microsoft.Azure.Commands.RecoveryServices.SiteRecovery
+{
+    /// <summary>
+    ///     Sets the Recovery Services vault context to be used for subsequent Azure Site Recovery operations in the current PowerShell session.
+    /// </summary>
+    [Cmdlet(
+        VerbsCommon.Set,
+        "AzureRmRecoveryServicesAsrVaultContext",
+        DefaultParameterSetName = ASRParameterSets.ARSVault,
+        SupportsShouldProcess = true)]
+    [Alias(
+        "Set-ASRVaultContext",
+        "Set-ASRVaultSettings",
+        "Set-AzureRmRecoveryServicesAsrVaultSettings")]
+    [OutputType(typeof(ASRVaultSettings))]
+    public class SetAzureRmRecoveryServicesAsrVaultSettings : SiteRecoveryCmdletBase
+    {
+        /// <summary>
+        ///     Gets or sets the Recovery Services vault object corresponding to the Recovery Services vault.
+        /// </summary>
+        [Parameter(
+            ParameterSetName = ASRParameterSets.ARSVault,
+            Mandatory = true,
+            ValueFromPipeline = true)]
+        [ValidateNotNullOrEmpty]
+        public ARSVault Vault { get; set; }
+
+        /// <summary>
+        ///     ProcessRecord of the command.
+        /// </summary>
+        public override void ExecuteSiteRecoveryCmdlet()
+        {
+            base.ExecuteSiteRecoveryCmdlet();
+
+            if (this.ShouldProcess(
+                this.Vault.Name,
+                VerbsCommon.Set))
+            {
+                  this.SetARSVaultContext(this.Vault);
+            }
+        }
+
+        /// <summary>
+        ///     Set Azure Recovery Services Vault context.
+        /// </summary>
+        private void SetARSVaultContext(
+            ARSVault arsVault)
+        {
+            try
+            {
+                using (var powerShell = System.Management.Automation.PowerShell.Create())
+                {
+                    VaultExtendedInfoResource vaultExtendedInfo = null;
+
+                    try
+                    {
+                        vaultExtendedInfo = this.RecoveryServicesClient
+                        .GetVaultExtendedInfo(this.Vault.ResourceGroupName, this.Vault.Name);
+                    }
+                    catch (CloudException ex)
+                    {
+                        // To Handle the case when cert is not uploaded to RecoveryServices and usage api is returning null.
+                        if (ex.Response.StatusCode.CompareTo(HttpStatusCode.NotFound) == 0)
+                        {
+                            var result = powerShell
+                                .AddCommand("Get-AzureRmRecoveryServicesVaultSettingsFile")
```

Comment:
@viverm why are we doing this? We shouldn't be creating a new PowerShell instance to call a cmdlet from a separate module that the user may or may not have installed. If we hit this exception, we should throw with a detailed error message that lets the user know they should install the `AzureRM.RecoveryServices` module and run the `Get-AzureRmRecoveryServicesVaultSettingsFile` with the provide `ARSVault`.

Reply:
Right  now  ASR has tight dependency with AzureRM.RecoveryServices module. The input for this cmdlet  is output of AzureRM.RecoveryServices. So the assumption here is if user is calling this cmdlet he already has the required module installed.

Will add support in next review where user can pass resourceId of vault instead of different module object.
@cormacpayne

Reply:
@viverm regardless of if ASR has a tight dependency on RecoveryServices, we cannot call a cmdlet from a separate module here, especially if it's going to force us to create a new PowerShell session to do so. In the short-term, we have a couple of options:

(1) Since ASR references the RecoveryServices project, you can reference the backend code that this cmdlet uses to make this call
(2) Write a message that tells the user to install the AzureRM.RecoveryServices module and run the listed cmdlet

In the long-term, please refer to the email I sent about consolidating all RecoveryServices modules into one.

Also, why are we adding this cmdlet when it already exists [here](https://github.com/AsrOneSdk/azure-powershell/blob/f79d00b6be470142c892f8c2c0a51c33afb4bf86/src/ResourceManager/RecoveryServices.SiteRecovery/Commands.RecoveryServices.SiteRecovery/Vault/SetAzureRmRecoveryServicesAsrVaultSettings.cs)? I don't understand why we are re-adding this cmdlet.

Reply:
@cormacpayne  looked at the mail .

Putting the message to user for now 
`
Message: For newly created vault download vault creds file and retry.
Recommended Action: Install module AzureRm.RecoveryServices if not installed. And execute below cmdlets
$path = Get-AzureRmRecoveryServicesVaultSettingsFile -Vault $Vault
Import-AzureRmRecoveryServicesAsrVaultSettingsFile -Path $path.filepath
`
cmdlet already Exists just renamed the file to new to match cmdlet name instead of alias.